src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderIndividualScheduleRepository.java

```diff
@@ -1,17 +1,20 @@
 package uk.gov.justice.hmpps.prison.repository.jpa.repository;
 
+import jakarta.persistence.LockModeType;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Lock;
 import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderIndividualSchedule;
 
 import java.util.List;
 import java.util.Optional;
 
 public interface OffenderIndividualScheduleRepository extends JpaRepository<OffenderIndividualSchedule, Long> {
-
-    Optional<OffenderIndividualSchedule> findOneByParentEventId(Long parentEventId);
+    @NotNull
```

Comment:
why do you need not null with an optional?  (sorry very rusty on the java side!)